### ISSUE TITLE
TypeScript errors in web-mercator-viewport.d.ts

### ISSUE THREAD

**Issue by danvk** [uber-web/math.gl#147]
I get these errors from `tsc` when I install `@math.gl/web-mercator`:

```
$ tsc --noEmit
node_modules/@math.gl/web-mercator/dist/es5/web-mercator-viewport.d.ts:144:33 - error TS7031: Binding element 'lngLat' implicitly has an 'any' type.

144   getMapCenterByLngLatPosition({lngLat, pos}): number[];
                                    ~~~~~~

node_modules/@math.gl/web-mercator/dist/es5/web-mercator-viewport.d.ts:144:41 - error TS7031: Binding element 'pos' implicitly has an 'any' type.

144   getMapCenterByLngLatPosition({lngLat, pos}): number[];
                                            ~~~

node_modules/@math.gl/web-mercator/dist/es5/web-mercator-viewport.d.ts:147:23 - error TS7031: Binding element 'lngLat' implicitly has an 'any' type.

147   getLocationAtPoint({lngLat, pos}): number[];
                          ~~~~~~

node_modules/@math.gl/web-mercator/dist/es5/web-mercator-viewport.d.ts:147:31 - error TS7031: Binding element 'pos' implicitly has an 'any' type.

147   getLocationAtPoint({lngLat, pos}): number[];
                                  ~~~

node_modules/@math.gl/web-mercator/dist/es5/web-mercator-viewport.d.ts:159:13 - error TS7006: Parameter 'bounds' implicitly has an 'any' type.

159   fitBounds(bounds, options?: {padding?: number; offset?: [number, number]}): WebMercatorViewport;
                ~~~~~~


Found 5 errors.
```

Looking at [that file](https://cdn.jsdelivr.net/npm/@math.gl/web-mercator@3.1.4/dist/es5/web-mercator-viewport.d.ts) there are, indeed, some bits that don't really look like TypeScript. I'm also seeing this problem in the same file in 3.2.0-alpha.2.

**Comment by ibgreen** [comment]
This should only affect 3.2 alpha releases, as typescript support is in development for 3.2. If you are seeing this in 3.1.x then let us know, as that means that something has been merged by mistake.

For 3.2-alpha we are indeed adding typescript type definitions but we are not actually typescript users ourselves so it is almost expected that we get some things wrong.

From your error it looks like we just need to add specifications for a few parameters?

Are you able to put up a PR for these errors or provide a bit more detail so that we can get a fix out quickly?

**Comment by Pessimistress** [comment]
@ibgreen Looks like the d.ts files do not exist in 3.1-release source, but somehow got into the transpiled directories. I will investigate and re-publish.

**Comment by danvk** [comment]
Thanks @Pessimistress!

@ibgreen it's really a judgment call, but the [official recommendation][1] is to bundle types "if your package is written in TypeScript." The reason is that hand-written typings are much more likely to have bugs, and it's much harder for TS users to work around bugs in bundled type declarations than it is to work around issues in types coming from DefinitelyTyped.

If you do want to bundle them, at the very least you should run `tsc` over your typings, and ideally a type declaration testing tool like [dtslint][2] or [tsd][3] as well.

[1]: https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package
[2]: https://github.com/microsoft/dtslint
[3]: https://github.com/SamVerschueren/tsd

**Comment by ibgreen** [comment]
> it's really a judgment call, but the official recommendation is to bundle types "if your package is written in TypeScript.

Then they should not have created the `d.ts` syntax, which is clearly intended for this use case ;) Also big libraries like THREE.js uses this and it seems to work well for them.

> If you do want to bundle them, at the very least you should run tsc over your typing

Yes of course. We do run `tsc --noEmit` on both the source and our tests without errors and warnings, so this was probably a just a result of type definition files from a newer version getting mixed with the built source of an older version.

Thanks for the thoughts and no worries, I was just trying to see if I could get a typescript user of our libraries interested in helping out, we can absolutely handle this.

**Comment by Pessimistress** [comment]
Published 3.1.5

**Comment by danvk** [comment]
btw @ibgreen @Pessimistress if you guys feel inspired to mention `math.gl` in response to my [Stack Overflow question](https://stackoverflow.com/questions/61719119/equivalent-of-map-cameraforbounds-without-a-mapbox-gl-map-instance) about calculating viewports without a map, I'd be happy to accept—it's solved the problem for me!
